doc/api/async_hooks.md

```diff
@@ -973,6 +973,15 @@ chains. It allows storing data throughout the lifetime of a web request
 or any other asynchronous duration. It is similar to thread-local storage
 in other languages.
 
+`AsyncLocalStorage` doesn't provide any new functionality that Async Hooks
+doesn't already provide: anything implemented with `AsyncLocalStorage` can be
+implemented with Async Hooks instead. The goal of `AsyncLocalStorage` is to
+provide a more performant and memory safe way to share state accross
```

Comment:
@vdeturckheim I'd argue to keep the `more` in `more performant` because, although there may be a more performant raw async hook implementation, the *intent* of AsyncLocalStorage is to be more performant.

Reply:
Agreed with @brillout there. Also, using `AsyncLocalStorage` eliminates the cognitive burden of needing to understand _how_ and _why_ the particulars of this implementation are faster. The advantage of avoiding the destroy hook is very non-obvious for one not familiar with the implications.